backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/chipgroup/single/internal/BpkSingleSelectChipGroupImpl.kt

```diff
@@ -59,29 +61,72 @@ internal fun BpkSingleSelectChipGroupImpl(
                 horizontalArrangement = Arrangement.spacedBy(BpkSpacing.Md),
             ) {
                 itemsIndexed(items = chips) { index, chip ->
-                    ChipItem(chip, index == selectedIndex, style) {
-                        onItemClicked.invoke(chips[index])
-                    }
+                    BehaviouralChipItem(
+                        behaviouralEventWrapper = behaviouralEventWrapper,
+                        chip = chip,
+                        selected = selectedIndex == index,
+                        style = style,
+                        onItemClicked = { onItemClicked(chips[index]) },
+                    )
                 }
             }
         }
 
         BpkSingleChipGroupType.Wrap -> {
             FlowRow(
-                modifier = modifier.selectableGroup().padding(contentPadding),
+                modifier = modifier
+                    .selectableGroup()
+                    .padding(contentPadding),
                 horizontalArrangement = Arrangement.spacedBy(BpkSpacing.Md),
                 verticalArrangement = Arrangement.spacedBy(BpkSpacing.Md),
             ) {
                 chips.forEachIndexed { index, chip ->
-                    ChipItem(chip, index == selectedIndex, style) {
-                        onItemClicked.invoke(chips[index])
-                    }
+                    BehaviouralChipItem(
+                        behaviouralEventWrapper = behaviouralEventWrapper,
+                        chip = chip,
+                        selected = selectedIndex == index,
+                        style = style,
+                        onItemClicked = { onItemClicked(chips[index]) },
+                    )
                 }
             }
         }
     }
 }
 
+@Composable
+private fun BehaviouralChipItem(
+    behaviouralEventWrapper: BpkBehaviouralEventWrapper?,
+    chip: BpkSingleChipItem,
+    selected: Boolean,
+    style: BpkChipStyle,
+    onItemClicked: () -> Unit,
+    modifier: Modifier = Modifier,
+) {
+    if (behaviouralEventWrapper != null) {
+        behaviouralEventWrapper(chip, Modifier) {
```

Comment:
the modifier should be passed here, rather than the chip item

Reply:
nice catch